packages/ethereum/contracts/HoprNetworkRegistry.sol

```diff
@@ -75,22 +89,52 @@ contract HoprNetworkRegistry is Ownable {
    * @notice It allows msg.sender to update registered node address.
    * @param hoprAddress Hopr nodes id. e.g. 16Uiu2HAmHsB2c2puugVuuErRzLm9NZfceainZpkxqJMR6qGsf1x1
    */
-  function selfRegister(string calldata hoprAddress) external mustBeEnabled returns (bool) {
+  function selfRegister(string memory hoprAddress) external mustBeEnabled returns (bool) {
+    require(bytes(hoprAddress).length > 0, 'HoprNetworkRegistry: HOPR node multiaddress must not be empty');
```

Comment:
Could we, sensibly, verify that the address is a valid address? Otherwise accounts could register bogus addresses.

Reply:
I'd put the validation off-chain to save gas and to leverage existing validation function(?) in `Multiaddr` or just a check on base58-encoding